kolibri/core/content/serializers.py

```diff
@@ -122,6 +124,52 @@ class Meta:
         )
 
 
+class PublicChannelSerializer(KolibriModelSerializer):
+    matching_tokens = serializers.SerializerMethodField('match_tokens')
+    language = serializers.SerializerMethodField()
+    included_languages = serializers.SerializerMethodField()
+    total_resource_count = serializers.SerializerMethodField()
+    icon_encoding = serializers.SerializerMethodField()
+    published_size = serializers.SerializerMethodField()
+    last_published = serializers.SerializerMethodField()
+    public = serializers.SerializerMethodField()
+
+    def get_public(self, instance):
+        return True
+
+    def get_language(self, instance):
+        if instance.root.lang is None:
+            return None
+
+        return instance.root.lang.lang_code
+
+    def get_included_languages(self, instance):
+        channel_nodes = ContentNode.objects.filter(channel_id=instance.id)
+        return list(channel_nodes.order_by('lang').values_list('lang', flat=True).distinct())
+
+    def get_total_resource_count(self, instance):
+        channel_nodes = ContentNode.objects.filter(channel_id=instance.id)
+        return channel_nodes.filter(available=True).exclude(kind=content_kinds.TOPIC).count()
+
+    def get_icon_encoding(self, instance):
+        return instance.thumbnail
+
+    def get_published_size(self, instance):
+        channel_nodes = ContentNode.objects.filter(channel_id=instance.id)
+        return _total_file_size(_files_for_nodes(channel_nodes).filter(available=True))
```

Comment:
The calculated fields above all look pretty expensive, which is worrying -- could we instead precalculate some of this stuff during the metadata import process?

(can discuss -- perhaps looping @rtibbles in also)

Reply:
If were are doing this only over available files, would have to be during annotation, I think?

Reply:
> Add fields to ChannelMetadata which will hold this data and probably add another method during annotation to fill in those fields, such as calculate_fields?

Sounds right to me. Let's get the merged in so we're not blocking, but seems important to then do the optimizations before 0.11 ships.